File: artifacts/api-server/internal/store/audit.go
package store

import (
	"context"
	"encoding/json"
)

// WriteAudit records an audit-log entry. It never returns an error to the
// caller in a way that should abort the primary operation; callers may ignore
// the result. (Mirrors the original resilient audit service.)
func (s *Store) WriteAudit(ctx context.Context, userID *string, action, entityType string, entityID *string, metadata map[string]any, ip *string) error {
	var meta []byte
	if metadata != nil {
		meta, _ = json.Marshal(metadata)
	}
	_, err := s.pool.Exec(ctx, `
		INSERT INTO "AuditLog" (id, "userId", action, "entityType", "entityId", metadata, "ipAddress", "createdAt")
		VALUES ($1, $2, $3, $4, $5, $6, $7, now())`,
		NewID(), userID, action, entityType, entityID, meta, ip)
	return err
}

File: attached_assets/_extracted/Competency-Platform-main/src/server/services/audit.ts
import type { Prisma, PrismaClient } from "@prisma/client";

export interface AuditEntry {
  userId?: string | null;
  action: string;
  entityType?: string;
  entityId?: string;
  metadata?: Prisma.InputJsonValue;
  ipAddress?: string | null;
}

/**
 * Writes an audit-log row. Never throws into the caller's path — auditing must not
 * break the operation it records; failures are logged to the server console.
 */
export async function writeAudit(
  db: PrismaClient,
  entry: AuditEntry,
): Promise<void> {
  try {
    await db.auditLog.create({
      data: {
        userId: entry.userId ?? null,
        action: entry.action,
        entityType: entry.entityType,
        entityId: entry.entityId,
        metadata: entry.metadata,
        ipAddress: entry.ipAddress ?? null,
      },
    });
  } catch (err) {
    console.error("[audit] failed to write audit log", err);
  }
}
